yarn_api_client/resource_manager.py

```diff
@@ -13,38 +13,34 @@ class ResourceManager(BaseYarnAPI):
     scheduler information, information about nodes in the cluster,
     and information about applications on the cluster.
 
-    If `address` argument is `None` client will try to extract `address` and
-    `port` from Hadoop configuration files.  If both `address` and `alt_address`
-    are provided, the address corresponding to the ACTIVE HA Resource Manager will
+    If `service_endpoint` argument is `None` client will try to extract it from
+    Hadoop configuration files.  If both `address` and `alt_address` are
+    provided, the address corresponding to the ACTIVE HA Resource Manager will
     be used.
 
-    :param str address: ResourceManager HTTP address
-    :param int port: ResourceManager HTTP port
-    :param str alt_address: Alternate ResourceManager HTTP address for HA configurations
-    :param int alt_port: Alternate ResourceManager HTTP port for HA configurations
+    :param List[str] service_endpoints: List of ResourceManager HTTP(S)
+    addresses
     :param int timeout: API connection timeout in seconds
-    :param boolean kerberos_enabled: Flag identifying is Kerberos Security has been enabled for YARN
+    :param AuthBase auth: Auth to use for requests configurations
+    :param boolean verify: Either a boolean, in which case it controls whether
+    we verify the server's TLS certificate, or a string, in which case it must
+    be a path to a CA bundle to use. Defaults to ``True``
     """
-    def __init__(self, address=None, port=8088, alt_address=None, alt_port=8088, timeout=30, kerberos_enabled=False):
-        if address is None:
+    def __init__(self, service_endpoints=None, timeout=30, auth=None, verify=True):
+        active_service_endpoint = None
+        if not service_endpoints:
             self.logger.debug('Get configuration from hadoop conf dir: {conf_dir}'.format(conf_dir=CONF_DIR))
-            address, port = get_resource_manager_host_port()
+            active_service_endpoint = get_resource_manager_host_port(timeout)
         else:
-            if alt_address:  # Determine active RM
-                if not check_is_active_rm(address, port):
-                    # Default is not active, check alternate
-                    if check_is_active_rm(alt_address, alt_port):
-                        address, port = alt_address, alt_port
+            for endpoint in service_endpoints:
+                if check_is_active_rm(endpoint, timeout):
+                    active_service_endpoint = endpoint
+                    break
 
-        super(ResourceManager, self).__init__(address, port, timeout, kerberos_enabled)
-
-    def get_active_host_port(self):
```

Comment:
Why is this removed?  This is used by clients to determine which RM is being used within an HA configuration.  It should be renamed to `get_active_endpoint` and return the endpoint which could be constructed using `self.service_uri.to_url(None)`.

Reply:
It seems to be not useful, as it's possible to take out the base class members service_endooint_uri

Reply:
This will be used by applications that pass in multiple endpoints but then need to know (for logging and troubleshooting purposes) which of those endpoints is the actual location for where things are happening.

Reply:
Brought it back